jcl/src/java.base/share/classes/java/lang/Class.java

```diff
@@ -2153,6 +2152,33 @@ String getPackageName() {
 	return getNonArrayClassPackageName(this);
 }
 
+/**
+ * Answers the name of the package to which the receiver belongs.
+ * For example, Object.class.getPackageName() returns "java.lang".
+ * Returns "java.lang" if this class represents a primitive type or void,
+ * and the element type's package name in the case of an array type.
+ *
+ * @return String the receiver's package name
+ *
+ * @see			#getPackage
+ */
+/*[IF Sidecar19-SE]*/
+public
+/*[ENDIF] Sidecar19-SE */
+String getPackageName() {
+	String packageName = packageNameString;
+	if (null == packageName) {
+		synchronized (this) {
```

Comment:
Why is this synchronized? If there is a race condition the result is always the same

Reply:
Yes, we do not need this to be synchronized, I have made the change in https://github.com/eclipse-openj9/openj9/compare/32da8a53a24fc33450f4215c896c382ca89f5cac..f34d5c7b9557fb792d9ae874a3de46bf15342526